src/points/saga.ts

```diff
@@ -0,0 +1,61 @@
+import { call, put, select, spawn, takeLeading } from 'typed-redux-saga'
+import { getHistoryStarted, getHistorySucceeded, getHistoryError } from 'src/points/slice'
+import { safely } from 'src/utils/safely'
+import networkConfig from 'src/web3/networkConfig'
+import { GetHistoryResponse } from 'src/points/types'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { getPointsHistoryNextPageUrlSelector } from 'src/points/selectors'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'Points/saga'
+
+export async function fetchHistory(
+  address: string,
+  url?: string | null
+): Promise<GetHistoryResponse> {
+  const response = await fetchWithTimeout(
+    url ?? `${networkConfig.getPointsHistoryUrl}?` + new URLSearchParams({ address }),
+    {
+      method: 'GET',
+    }
+  )
+  if (response.ok) {
+    return response.json() as Promise<GetHistoryResponse>
+  } else {
+    throw new Error(await response.text())
+  }
+}
+
+export function* getHistory({ payload: params }: ReturnType<typeof getHistoryStarted>) {
+  const walletAddress = yield* select(walletAddressSelector)
+  if (!walletAddress) {
+    Logger.error(TAG, 'No wallet address found when fetching points history')
+    yield* put(getHistoryError())
+    return
+  }
+
+  const url = params.fromPage ? yield* select(getPointsHistoryNextPageUrlSelector) : undefined
+
+  try {
+    const history = yield* call(fetchHistory, walletAddress, url)
+    yield* put(
+      getHistorySucceeded({
+        appendHistory: params.fromPage,
+        newPointsHistory: history.data,
+        nextPageUrl: history.hasNextPage ? history.nextPageUrl : null,
```

Comment:
null will be stored for the nextPageUrl, so i assume that we'll take that on the front end to mean we should not fetch more?

Reply:
Actually, I had meant to handle this from within the `getHistory` method itself, e.g., silently succeeding without fetching if a refresh is requested but no page information is available; I don't think callers should be expected to check that beforehand.